arangod/Agency/Agent.cpp

```diff
@@ -348,18 +350,41 @@ priv_rpc_ret_t Agent::recvAppendEntriesRPC(
     return priv_rpc_ret_t(false,t);
   }
 
+  size_t nqs = payload.length();
+  if(nqs > 0 && !payload[0].get("readDB").isNone()) {
```

Comment:
Is this clever? Assume we are at index 2000 but our last snapshot is at 1000. If we now get an appendEntriesRPC with a snapshot at 1500 and 99 more log entries, should we then not check if our log fits to the snapshot and data and not change anything if everything is OK? With this, we are throwing away a huge performance improvement which we have already implemented and which works. I think we must be more clever here and perform the check even when we receive a snapshot. Remember that snapshotting could be done differently in leader and follower!

Reply:
will leave as is for now.